test/erlfmt_SUITE.erl

```diff
@@ -956,6 +964,27 @@ smoke_test_stdio_with_pragma(Config) when is_list(Config) ->
     {ok, Expected} = file:read_file(Path),
     ?assertEqual(Expected, unicode:characters_to_binary(Formatted)).
 
+smoke_test_stdio_insert_pragma_without(Config) when is_list(Config) ->
+    Formatted = os:cmd("echo '-module(nopragma).' | " ++ escript() ++ " - --insert-pragma"),
+    Expected =
+        "%% @format\n"
+        "\n"
+        "-module(nopragma).\n",
+    ?assertEqual(Expected, Formatted).
+
+smoke_test_stdio_insert_pragma_with(Config) when is_list(Config) ->
+    DataDir = ?config(data_dir, Config),
+    Path = filename:join(DataDir, "pragma.erl"),
+    Formatted = os:cmd("cat " ++ Path ++ " | " ++ escript() ++ " - --insert-pragma"),
+    {ok, Expected} = file:read_file(Path),
+    ?assertEqual(Expected, unicode:characters_to_binary(Formatted)).
+
+smoke_test_stdio_insert_and_require_pragma(Config) when is_list(Config) ->
+    DataDir = ?config(data_dir, Config),
+    Path = filename:join(DataDir, "pragma.erl"),
+    ErrorString = os:cmd("cat " ++ Path ++ " | " ++ escript() ++ " - --insert-pragma --require-pragma"),
+    ?assert(string:find(ErrorString, "Cannot use both --insert-pragma and --require-pragma") =/= nomatch).
```

Comment:
Is there a reason we're testing this specifically through stdio instead of the regular smoke test infrastructure we have?

Reply:
No reason specifically, it was just easy.  I will move most to snapshot testing, I assume that is what you mean with regular smoke test infrastructure.

Reply:
Hmmm, looking at the snapshot setup again, I think I picked the stdio version, because it is the only one that invokes the command with `os:cmd` and I want to test everything including the argument parsing.